src/app/ui/editor/pixels_movement.cpp

```diff
@@ -492,10 +494,34 @@ void PixelsMovement::moveImage(const gfx::PointF& pos, MoveModifier moveModifier
       }
 
       // Snap to grid when resizing tilemaps
-      if (m_site.tilemapMode() == TilemapMode::Tiles) {
-        gfx::Rect gridBounds = m_site.gridBounds();
-        a = gfx::PointF(snap_to_grid(gridBounds, gfx::Point(a), PreferSnapTo::BoxOrigin));
-        b = gfx::PointF(snap_to_grid(gridBounds, gfx::Point(b), PreferSnapTo::BoxOrigin));
+      if (tilemapMode) {
+        // Most right-bottom pixel adjust
+        a.x = a.x - (a.x > b.x? 1 : 0);
+        a.y = a.y - (a.y > b.y? 1 : 0);
+        b.x = b.x - (a.x <= b.x? 1 : 0);
+        b.y = b.y - (a.y <= b.y? 1 : 0);
+        // Snap to tilemap grid
+        gfx::RectF alignedRect(m_site.grid().alignBounds(gfx::RectF(a, gfx::SizeF(1, 1)) |
+                                                         gfx::RectF(b, gfx::SizeF(1, 1))));
+        // We want to keep a rectangle with negative width or
+        // height (to know that it was flipped).
+        // TO DO: implement flips on tilemaps.
+        if (a.x <= b.x) {
+          a.x = alignedRect.x;
+          b.x = alignedRect.x2();
+        }
+        else {
+          a.x = alignedRect.x2();
+          b.x = alignedRect.x;
+        }
+        if (a.y <= b.y) {
+          a.y = alignedRect.y;
+          b.y = alignedRect.y2();
+        }
+        else {
+          a.y = alignedRect.y2();
+          b.y = alignedRect.y;
+        }
```

Comment:
I'm not sure, but I feel like if this should be moved to some function. Looking at the old code, there is an snap_to_grid() function, so maybe we should create a new one for doing this?

Reply:
I'm leaving this as is, because scaling on the tilemap will be an open problem:
- I think inverted tilemaps (scale factor < 1) will not be supported yet (so this code to keep the negative factor is not needed at the moment).
- I think scaling on tilemaps will be good as an expansion of the pattern instead of repeating the last column/row of tiles in the selection.
We need to talk to David about these points.

Reply:
As we talked to David. The scaling on the tilemap is as a 9-slice scale (with the last row/column as a frame and the center as a repeated pattern):
Example:
<img width="175" alt="Screenshot 2022-11-10 at 16 15 19" src="https://user-images.githubusercontent.com/32165475/201186138-b43410b4-7ce4-4c0e-9fe0-1594283e8ea8.png">

Finally, I used `snap_to_grid()` function. Thanks for the advice.

@martincapello could you review/test one more time this PR